### ISSUE TITLE
Add buffers with regexp "wrong argument type: stringp #<buffer>"

### ISSUE THREAD

**Issue by mkschleg** [Bad-ptr/persp-mode.el#105]
When using the function `persp-add-buffers-by-regexp` there is an error when doing the regexp match. Instead of passing a list of buffers to `persp-do-buffer-list-by-regexp`, the function expects a list of names. To fix we could give it a list of names *or* modify the list of buffers to a list of names in `persp-do-buffer-list-by-regexp`.

I did the former locally, but wanted to get feedback on how you would like to handle it before I submit a PR.

**Comment by Bad-ptr** [comment]
Something like this?

```elisp

(defun* persp-do-buffer-list-by-regexp (&key func regexp blist noask
                                             (rest-args nil rest-args-p))
  (interactive)
  (unless func
    (let ((fs (completing-read "What function to apply: " obarray 'functionp t)))
      (when (and fs (not (string= fs "")))
        (setq func (read fs)))))
  (when func
    (unless regexp
      (setq regexp (read-regexp "Regexp: ")))
    (when regexp
      (unless blist
        (setq blist (eval (read--expression "Buffer list expression: " "nil"))))
      (when blist
        (unless rest-args-p
          (setq rest-args (read--expression "Rest arguments: " "nil")))

;; ------- THIS PART

        (setq blist
              (remove-if-not
               (apply-partially #'persp-string-match-p regexp)
               (mapcar #'get-buffer blist)
               :key #'buffer-name))

;; ------------------------

        (when (and blist
                   (or noask (y-or-n-p (format "Do %s on these buffers:\n%s?\n"
                                               func
                                               (mapconcat #'buffer-name blist ", ")))))
          (mapcar #'(lambda (b) (apply func b rest-args)) blist))))))
```

**Comment by Bad-ptr** [comment]
Must be fixed in 1e2a67c0d14547b579cc0861a4a064eb587e37b5
